### PR TITLE
fix: change devicePixelRatio to hardwareScalingLevel and fix thinInstance pick index

### PR DESCRIPTION
**Pull request by zhangyahan**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15332/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15332/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15332/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15332/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15332/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15332/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15332/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15332/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15332/merge/testResults/webgl2playwright/index.html

File: packages/dev/core/src/Collisions/gpuPicker.ts [hunk 1]
```diff
@@ -185,28 +204,31 @@ export class GPUPicker {
             const r = (id & 0xff0000) >> 16;
             const g = (id & 0x00ff00) >> 8;
             const b = (id & 0x0000ff) >> 0;
-            this._idMap[id] = index;
-            id++;
 
             if (mesh.hasThinInstances) {
-                const colorData = this._generateColorData((mesh as Mesh).thinInstanceCount, id, index, r, g, b, (i, id) => {
+                const colorData = this._generateThinInstanceColorData((mesh as Mesh).thinInstanceCount, id, index, r, g, b, (i, id) => {
```
Comment (deltakosh) [inline packages/dev/core/src/Collisions/gpuPicker.ts:209]: Why this change?

Comment (zhangyahan) [reply packages/dev/core/src/Collisions/gpuPicker.ts:209]: I found during debugging that the index returned during picking for thin instances is incorrect. Therefore, I believe that thin instances should not handle themselves, whereas instantiated models do have their own handling. So, I added a new _generateThinInstanceColorData method to differentiate between the handling logic for thin instances and instantiated models without disrupting the existing logic. This is why hasThinInstances does not increment the ID, but directly adds the thinInstanceCount. In the _generateThinInstanceColorData method, it ignores the existence of the model itself and directly processes the thin instances, ensuring the index is correct during picking.

My English is not very good, so I'm not sure if you understand what I'm saying.

Comment (deltakosh) [reply packages/dev/core/src/Collisions/gpuPicker.ts:209]: Do you have a playground for me to test? I understand what you mean but I'm not sure to understand why we need it

Comment (zhangyahan) [reply packages/dev/core/src/Collisions/gpuPicker.ts:209]: in here: #JT0999#2，please see in host.

Comment (zhangyahan) [reply packages/dev/core/src/Collisions/gpuPicker.ts:209]: @deltakosh Have you tested it?

Comment (deltakosh) [reply packages/dev/core/src/Collisions/gpuPicker.ts:209]: Ok this is great! My only ask would be to factorize the code between `_generateThinInstanceColorData` and `_generateColorData` as most of the code is similar

Comment (zhangyahan) [reply packages/dev/core/src/Collisions/gpuPicker.ts]: I'm not sure how to factorize, so I modified the parameter list of the _generateThinInstanceColorData function. Make it look different from the _generaColorData function.

**Review by deltakosh — APPROVED** [review]
